Разбираем склад Amazon BJS2-79877

DimonDem123 opened the pull request:
Хорошего времени суток

DimonDem123 committed:
BJS2-79242
Сделал проверки на пустные значения

DimonDem123 committed:
BJS2-79721
Потыкал проверил все работает коректно. Все мои тесты прошла
Хорошего дня

zaqqaz226 commented on src/main/java/school/faang/bjs279877/Category.java:
```diff
@@ -0,0 +1,21 @@
+package school.faang.bjs279877;
+
+public enum Category {
```
Добавь в название что то, чтобы было понятно категория чего. Категория товара, магазина или еще чего то)

DimonDem123 replied:
Ok

zaqqaz226 commented on src/main/java/school/faang/bjs279877/Main.java:
```diff
@@ -0,0 +1,27 @@
+public class Main {
+    public static void main(String[] args) {
+        ProductManager dataBase = new ProductManager();
+        //Попытка вывести все значения при пустых значения
```
Не оставляем комментарии в коде. В идеале мы стремимся к самодокументируемому коду. Это когда за счет названия классов, переменных, методов и так понятно, что там происходит.

zaqqaz226 commented on src/main/java/school/faang/bjs279877/ProductManager.java:
```diff
@@ -0,0 +1,112 @@
+            log.info("База данных пока пустая.");
+            return true;
+        }
+        return false;
```
Перенеси в else. Чтобы у нас сохранилась строгая инвариантность: либо то, либо это. Да и читать станет проще )

DimonDem123 replied:
Соглашусь. Так действительно проще. Постараюсь так и делать. Если не получиться, я просто человек и у меня лапки

zaqqaz226 commented on src/main/java/school/faang/bjs279877/ProductManager.java:
```diff
@@ -0,0 +1,112 @@
+    private int id = 0;
+
+    private void add(Category category, String name) {
+        Product product = new Product(id++, name, category);
```
Именно в данном случае не нравится id++ при инициализации. Я бы это вынес вне инициализации и просто оставил как строку в конце или начале метода add.

DimonDem123 replied:
Ок

zaqqaz226 commented on src/main/java/school/faang/bjs279877/ProductManager.java:
```diff
@@ -0,0 +1,112 @@
+        } else {
+            for (Product product : products) {
+                if (product.getName().equals(name) && product.getCategory().equals(category)) {
+                    System.out.println("Товар с таким именем уже существует в категории " + category);
```
Тоже в логирование переделай

zaqqaz226 commented on src/main/java/school/faang/bjs279877/ProductManager.java:
```diff
@@ -0,0 +1,112 @@
+            for (Product product : entry.getValue()) {
+                System.out.println("- " + product.getName());
+            }
+            System.out.println();
```
А это зачем?)

DimonDem123 replied:
Чтобы была пустая стока между каталогами, для удобного просмотра вывода

zaqqaz226 replied:
Это не стоит того. В логах все равно все в одну кучу и просто вычитываем. Так что это лишний код, который не несет функциональность.

DimonDem123 committed:
BJS2-79877
Исправил вывод. Убрал коментарии добавил лог так где указали, может что-то не заметил, но смысл уловил для чего исправления
